app/routes.py

```diff
@@ -1,2 +1,259 @@
-from flask import Blueprint
+from flask import Blueprint, request, make_response
+from flask import jsonify
+from app import db, slack_key, slack_channel
+from app.models.task import Task
+from app.models.goal import Goal
+import datetime
+import requests
 
+# class Task routes
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+
+@tasks_bp.route("", methods=["POST", "GET"], strict_slashes=False)
+def handle_tasks():
+
+    if request.method == "POST":
+
+        request_body = request.get_json()
+
+        if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+            return jsonify({
+                "details": "Invalid data"
+            }), 400
+        
+        new_task = Task(title=request_body["title"],
+                        description=request_body["description"],
+                        completed_at=request_body["completed_at"])
+
+        db.session.add(new_task)
+        db.session.commit()
+
+        return new_task.to_json(), 201
+
+    if request.method == "GET":
+
+        sort_query = request.args.get("sort")
+        title_query = request.args.get("title")
+
+        if sort_query == 'asc':
+            tasks = Task.query.order_by(Task.title)
+        elif sort_query == 'desc':
+            tasks = Task.query.order_by(Task.title.desc())
+        elif sort_query == 'id':
+            tasks = Task.query.order_by(Task.task_id)
+        elif title_query:
+            tasks = Task.query.filter_by(title=title_query)
+        else: 
+            tasks = Task.query.all()
+        
+        task_list = []
+
+        for task in tasks: 
+            task_list.append({
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": task.check_if_complete()
```

Comment:
Consider using your `to_json` method here. Currently your `to_json` method nests the dictionary, but if we removed that level of nesting in the method, it could be used in more places.